code/obj/item/flamethrower.dm

```diff
@@ -20,28 +30,133 @@ GETLINEEEEEEEEEEEEEEEEEEEEE
 	w_class = 4
 	var/mode = 1
 	var/processing = 0
-	var/operating = 0
-	var/throw_amount = 100
 	var/lit = 0	//on or off
-	var/base_temperature = 700
-	var/temp_loss_per_tile = 35
+	/// Set the projectile to make the reagents be this hot
+	var/base_temperature = FLAMER_DEFAULT_TEMP
+	/// Minimum temperature this thing'll set the reagents to
+	var/min_temperature = FLAMER_MIN_TEMP
+	/// Maximum temperature this thing'll set the reagents to
+	var/max_temperature = FLAMER_MAX_TEMP
+	/// Can this thing vary the temperature?
+	var/adjustable_temp = 1
+	/// How much chem to try to stuff into the projectile
+	var/amt_chem = FLAMER_DEFAULT_CHEM_AMT
+	/// Maximum volume this thing can shoot per shot
+	var/amt_chem_max = FLAMER_MAX_CHEM_AMT
+	/// Can this thing vary the amount of stuff loaded?
+	var/adjustable_chem_amt = 1
 	var/obj/item/tank/gastank = null
+	var/obj/item/fueltank = null // Honestly, anything with reagents'll do. Just needs sprites!
+	/// Can we swap out the tanks? Mainly so backpack flamers don't lose their flamer
+	var/swappable_tanks = 1
+	/// Is this a fancy combat flamethrower? Boosts melee damage, reduces spread
+	var/combat_flamer = 0
+	current_projectile = new/datum/projectile/special/shotchem
 	contraband = 5 //Heh
 	m_amt = 500
 	stamina_damage = 15
 	stamina_cost = 15
 	stamina_crit_chance = 1
 	move_triggered = 1
-
-	proc/get_reagsource(mob/user)
-		return null
+	/// Burstfire, but actually
+	use_shootloop = 1
+	refire_delay = 4 DECI SECONDS
+	burst_count = 3
+	spread_angle = 5
 
 	New()
 		..()
 		BLOCK_SETUP(BLOCK_LARGE)
 		setItemSpecial(null)
 
-/obj/item/flamethrower/assembled
+	/// Just check if there's a usable air and fuel tank
+	canshoot()
+		if(istype(src.gastank) && src.gastank.air_contents && istype(src.fueltank) && src.fueltank.reagents)
+			return TRUE
+
+	/// check for tank, pressure in tank, fuelltank, fuel in tank, and... then dump the stuff into it!
+	process_ammo(var/mob/user)
+		var/turf/T = get_turf(src)
+		var/datum/gas_mixture/T_env = T.return_air()
+		if(!src.fueltank)
+			boutput(user, "<span class='alert'>This [src] doesn't have a fuel source!</span>")
+			return FALSE
+		else if(!(src.fueltank in src.contents) && !(src.fueltank in user.get_equipped_items())) // Tank is loaded
+			boutput(user, "<span class='alert'>You need to either wear [src]'s fuel source or load it into the weapon!</span>")
+			return FALSE
+		else if(src.fueltank?.reagents.total_volume <= 0)
+			boutput(user, "<span class='alert'>This [src]'s fuel source is empty!</span>")
+			return FALSE
+		else if(T_env && src.gastank?.air_contents)
+			if(MIXTURE_PRESSURE(T_env) > MIXTURE_PRESSURE(gastank.air_contents))
+				boutput(user, "<span class='alert'>Not enough pressure in [src]'s gas tank to operate!</span>")
+				return FALSE
+		return TRUE
+
+		// else if(IN_RANGE(get_turf(src), get_turf(src.fueltank), 1))
+		// 	return TRUE
```

Comment:
?

Reply:
Must've missed that. And all the others. Removing!